Runtime/LLMChatHistory.cs

```diff
@@ -0,0 +1,139 @@
+/// @file
+/// @brief File implementing the LLMChatHistory.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace LLMUnity
+{
+    /// @ingroup llm
+    /// <summary>
+    /// Manages a single instance of a chat history.
+    /// </summary>
+    public class LLMChatHistory : MonoBehaviour
+    {
```

Comment:
Why is it defined as MonoBehaviour?
To expose the ChatHistoryFilename?

Reply:
Following the Unity-style component architecture. If someone wants to, say, create a UI piece that just needs a chat history and no associated LLM, they can do that using this as a separate component. Maybe down the line we integrate RAG capabilities to it (or have a RAG enabled chat history that extends the base LLMChatHistory). This design leaves us open to the ChatHistory being it's own entity independent of the LLM.

Presumably as features are added, more toggles/inputs may be exposed in the component editor UI as well.